File: client/src/models/Item.ts
type ItemJSON = {
  id?: number;
  name?: string;
  img?: string;
  price?: number;
  quantity?: number;
  unit?: string;
};
class Item {
  #id: number;
  name: string;
  img: string;
  price: number;
  quantity: number;
  unit: string;

  constructor(itmJSON: ItemJSON) {
    this.#id = itmJSON.id ?? 0;
    this.name = itmJSON.name ?? "";
    this.img = itmJSON.img ?? "";
    this.price = itmJSON.price ?? 0;
    this.quantity = itmJSON.quantity ?? 0;
    this.unit = itmJSON.unit ?? "";
  }
  get id(): number {
    return this.#id;
  }
}

export { type ItemJSON };
export default Item;

File: server/models/Item.go
package models

// ──────────────────────────────────────────────
type Item struct {
	ID           int           `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	Name         string        `gorm:"column:name;not null;unique" json:"name"`
	Img          string        `gorm:"column:img;not null" json:"img"`
	Price        int           `gorm:"column:price;not null" json:"price"`
	Quantity     uint          `gorm:"column:quantity;not null" json:"quantity"`
	Unit         string        `gorm:"column:unit;not null" json:"unit"`
	PackageItems []PackageItem `gorm:"foreignKey:ItemID" json:"-"`
	BuyItems     []BuyItem     `gorm:"foreignKey:ItemID" json:"buy_items"`
}

func (Item) TableName() string {
	return "item"
}
